test-runner-poc/lib/filters/mongodb_topology_filter.js

```diff
@@ -11,14 +12,36 @@
  * }
  */
 class MongoDBTopologyFilter {
-  constructor(options) {
-    this.runtimeTopology = options.runtimeTopology || 'single';
+  initializeFilter(callback) {
+    const mongoClient = new MongoClient(process.env.MONGODB_URI || 'mongodb://127.0.0.1:27017');
+    mongoClient.connect((err, client) => {
+      if (err) throw new Error(err);
```

Comment:
Why are we constructing an `Error` here?

Reply:
We wanted to stop the program and handle the error - before, it would skip over the error.

Reply:
You do not need to wrap the returned error in an error. If it exists, then it is already an `Error`.

More importantly: Throwing here will likely have a bad effect on the program, b/c you are no longer in a synchronous call stack. You will want to propagate the error on by passing it to the callback

```js
if (err) {
  callback(err);
  return;
}
```